flink-core/src/main/java/org/apache/flink/configuration/ConfigOption.java

```diff
@@ -195,6 +196,27 @@ public T defaultValue() {
 		return defaultValue;
 	}
 
+	/**
+	 * Checks whether this option has deprecated keys.
+	 * @return True if the option has deprecated keys, false if not.
+	 * @deprecated Replaced by {@link #hasFallbackKeys()}
+	 */
+	@Deprecated
+	public boolean hasDeprecatedKeys() {
+		return hasFallbackKeys();
+	}
+
+	/**
+	 * Gets the deprecated keys, in the order to be checked.
+	 * @return The option's deprecated keys.
+	 * @deprecated Replaced by {@link #fallbackKeys()}
+	 */
+	@Deprecated
+	public Iterable<String> deprecatedKeys() {
+		return fallbackKeys == EMPTY ? Collections.<String>emptyList() :
+			Arrays.stream(fallbackKeys).map(fallbackKey -> fallbackKey.getKey()).collect(Collectors.toList());
```

Comment:
```suggestion
			Arrays.stream(fallbackKeys)
			    .filter(FallbackKey::isDeprecated)
			    .map(FallbackKey::getKey)
			    .collect(Collectors.toList());
```

Reply:
I think you are right. Additionally, we may also need to change the implementation of the`hasDeprecatedKeys` method, right? Only based on the `fallbackKeys != EMPTY` condition, can not judge a config option has deprecated keys if the fallback key is not marked as deprecated.